src/shell/shell_misc.cpp

```diff
@@ -644,36 +644,20 @@ std::optional<std::string> DOS_Shell::GetEnvStr(const std::string_view entry) co
 	}
 }
 
-bool DOS_Shell::GetEnvNum(Bitu num, std::string& result) const
+std::vector<std::string> DOS_Shell::FullEnvironment() const
 {
+	auto full_env = std::vector<std::string>{};
+
 	char env_string[1024 + 1];
 	PhysPt env_read = PhysicalMake(psp->GetEnvironment(), 0);
-	do {
+	while(true) {
```

Comment:
`for (;;)`

Reply:
hopefully the most common or logical break-criteria can be put in the while loop and we can avoid both of them (`while(true)` and `for (;;)`.